editor/gui/editor_quick_open_dialog.cpp

```diff
@@ -795,6 +765,34 @@ void QuickOpenResultItem::highlight_item(bool p_enabled) {
 	queue_redraw();
 }
 
+void QuickOpenResultItem::draw_search_highlights() {
+	if (dirty_highlights) {
+		// When initially switching layouts, the new sub-item has not yet been positioned, so this
+		// delays finding and drawing highlights until after that happens.
```

Comment:
That sounds quite hacky. When is this needed?

Reply:
Maybe use RichTextLabel instead? Coloring with BBCodes would be more reliable.

Reply:
I was thinking about that option earlier on and I wasn't sure; it seems like RichTextLabel would be more expensive and I don't think a similar highlighting effect could be achieved (it looks like some bbcode implementations include `[mark]` or `[background]` but godot bbcode does not).

Reply:
Ehh IMO after trying it the custom label also isn't much better due to how godot handles the minimum size of labels with overrun handling behavior (w/o using expand, since the text here is centered). If they could size up to their parent container, as you might expect, rather than truncate at anything greater than the custom minimum size, this would be pretty clean, although it adds a bunch of calls that otherwise were handled in a batch in the current code.

Reply:
Sounds like your horizontal size flags are wrong. Make sure your Label uses FILL flag, and possibly EXPAND.

Reply:
The issue was with the centered expanded label I wasn't sure how to deal with the padding/offset but I just discovered `get_character_bounds`. It would be handy if Label just exposed the computed Rect2 for the entire text but oh well this is good enough with minimal magic values.

The other option I was exploring was centering the label in a container to get the non-expanded rect but turns out with overrun handling the min width is 1px.